src/main/java/at/hannibal2/skyhanni/data/MiningAPI.kt

```diff
@@ -81,4 +267,38 @@ object MiningAPI {
         cold = newCold
     }
 
+    private fun updateLocation() {
+        val newInGlacite = inGlaciteArea()
+        val newInDwarvenMines = inRegularDwarven()
+        val newInCrystalHollows = inCrystalHollows()
+        val newInCrimsonIsle = IslandType.CRIMSON_ISLE.isInIsland()
+        val newInEnd = IslandType.THE_END.isInIsland()
+        val newInSpidersDen = IslandType.SPIDER_DEN.isInIsland()
+
+        if (newInGlacite != inGlacite ||
+            newInDwarvenMines != inDwarvenMines ||
+            newInCrystalHollows != inCrystalHollows ||
+            newInCrimsonIsle != inCrimsonIsle ||
+            newInEnd != inEnd ||
+            newInSpidersDen != inSpidersDen
+        ) {
+
+            inGlacite = newInGlacite
+            inDwarvenMines = newInDwarvenMines
+            inCrystalHollows = newInCrystalHollows
+            inCrimsonIsle = newInCrimsonIsle
+            inEnd = newInEnd
+            inSpidersDen = newInSpidersDen
+
+            val entries = OreBlock.entries
+            val oreBlockList: List<OreBlock> = if (newInGlacite) entries.filter { it.checkArea.invoke() }
```

Comment:
here are many things weird:
it looks like we update this every tick, instead of every time the user changes the area?
why are there non mining islands?
why is every case doing the same thing?
you can remove .invoke,
you can use switch instead of if else chain